lib/std/os/test.zig

```diff
@@ -601,3 +601,7 @@ test "getrlimit and setrlimit" {
     const cpuLimit = try os.getrlimit(.CPU);
     try os.setrlimit(.CPU, cpuLimit);
 }
+
+test "os.sched_getaffinity" {
+    testing.expectError(error.NoThread, os.sched_getaffinity(std.math.maxInt(i32)));
```

Comment:
My assumption here is that we will never have as many threads as `maxInt(i32)` when running tests.    
But there is a good probability that I'm wrong/

Reply:
On windows and Osx, this is failing with `error: use of undeclared identifier 'cpu_set_t'` . 

Should I restrict this test to linux only?

Reply:
> On windows and Osx, this is failing with

Welcome to the world of cross-platform non-posix abstractions (and yet another opportunity to plug #6600):
* NetBSD has `pthread_setaffinity_np`/`pthread_getaffinity_np`
* FreeBSD has `cpuset_getaffinity`/`cpuset_setaffinity`
* macOS has no direct equivalent, `thread_policy_get`/`thread_policy_set` let you get/set the affinity tag for a given thread but that's it
* OpenBSD has no such API (cc @semarie)

> My assumption here is that we will never have as many threads as maxInt(i32) when running tests.

That's a pretty safe assumption (why are PIDs signed???), spawning that many threads requires an ungodly amount of memory heh

Reply:
I have skipped the test for non-linux OSes. I have also opened https://github.com/ziglang/zig/issues/6907 to implement `sched_getaffinity` for other OSes. I'll attempt to do that in a follow up PR, if no one else will have done so already.